src/Knp/Component/Pager/Paginator.php

```diff
@@ -53,9 +55,19 @@ public function setDefaultPaginatorOptions(array $options): void
 
     public function paginate($target, int $page = 1, int $limit = null, array $options = []): PaginationInterface
     {
+        if ($page <= 0) {
+            throw new PageNumberInvalidException(
+                sprintf('Invalid page number. Page: %d: $page must be positive non-zero integers', $page),
```

Comment:
"integer" must be singular here. Moreover, I'd move the error message inside the exception class.